infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -44,95 +68,231 @@ public class LogReplicationConfigManager {
     @Getter
     private final CorfuRuntime runtime;
 
-    @Getter
-    private LogReplicationConfig config;
+    private final CorfuStore corfuStore;
+
+    private final String localClusterId;
+
+    private long lastRegistryTableLogTail = Address.NON_ADDRESS;
 
     @Getter
     private ServerContext serverContext;
 
-    private long lastRegistryTableLogTail = Address.NON_ADDRESS;
+    // Map from a session to its corresponding config.
+    @Getter
+    private final Map<LogReplicationSession, LogReplicationConfig> sessionToConfigMap = new ConcurrentHashMap<>();
+
+    // Set of registered log replication subscribers.
+    @Getter
+    private final Set<ReplicationSubscriber> registeredSubscribers = new CopyOnWriteArraySet<>();
+
+    // Map from a logical group to all the Sinks it is targeting.
+    private final Map<String, Set<String>> groupSinksMap = new ConcurrentHashMap<>();
```

Comment:
`LogReplicationConfig` and its model-based inheritors are per session, and for LogicalGroupConfig it will have fields for which groups this config is associated with. For this field it's used by ConfigManager to generate/update config based on group-destination change.

Reply:
This seems related to Logical group use case only, right? Should we change groupSinksMap -> logicalGroupSinksMap?